code/framework/auditing/src/main/java/io/cattle/platform/iaas/api/auditing/AuditServiceImpl.java

```diff
@@ -140,8 +144,10 @@ private AuditEventType convertToAction(String method) {
     @Override
     public void logResourceModification(Object resource, Map<String, Object> data, AuditEventType eventType,
                                         String description, Long accountId, String clientIp) {
-        putInAsString(data, "resource", "Failed to convert resource to json.", resource);
-        publishEvent(auditLogDao.create(objectManager.getType(resource), parseId((String) ObjectUtils.getId(resource)),
+        if (data != null) {
+            putInAsString(data, "resource", "Failed to convert resource to json.", resource);
+        }
+        publishEvent(auditLogDao.create(objectManager.getType(resource), parseId(String.valueOf(ObjectUtils.getId(resource))),
                 data, null, accountId, null, eventType.toString(), null, null, description, clientIp));
```

Comment:
EventType that we put to publishEvent method, should be constructed similar to its API counterpart: resourceType.actionType. Example: service.update. So instead of eventType.toString(), we should use objectManager.getType(resource) + "." + eventType.toString()